src/tests/compiler/class/classDeclarationTests.ts

```diff
@@ -1347,4 +1348,64 @@ class Child extends Mixin(Base) {}
             doTest("class I {}\n\nclass J {}\n\nclass K {}", 1, "class I {}\n\nclass K {}");
         });
     });
+
+    describe(nameof<ClassDeclaration>(d => d.getStructure), () => {
+        function doTest(code: string, expectedStructure?: ClassDeclarationStructure) {
+            const {descendant, project} = getInfoFromTextWithDescendant<ClassDeclaration>(code, SyntaxKind.ClassDeclaration);
+            const structure = descendant.getStructure();
+            if (expectedStructure) {
+                expect(Object.assign({}, structure, {methods: undefined, properties: undefined, ctors: undefined})).to.contain(
+                    Object.assign({}, expectedStructure, {methods: undefined, properties: undefined, ctors: undefined}));
+            }
+            const sourceFile2 = project.createSourceFile("__tmp_sourceFile1.ts", "", {overwrite: true});
```

Comment:
You can just add to the same file as above and then compare the texts of the two classes.